### DISCUSSION TITLE
Why there is no hardware SPI available in u8g2setupc?

### DISCUSSION THREAD

**Discussion by Wosamarak** [olikraus/u8g2#2387] [Q&A] [answered]
When using the plain C initialisation I can't find any hint, how to work with hardware SPI.
[u8g2setupc](https://github.com/olikraus/u8g2/wiki/u8g2setupc)
I can only find *sw_spi* init functions.

Is there a way, to use hardware SPI with plain C, or do I have to migrate my project to CPP?

**Comment by olikraus** [comment, marked as answer]
> When using the plain C initialisation I can't find any hint, how to work with hardware SPI.

You need to add calls to your uC specific SPI subsytem:
https://github.com/olikraus/u8g2/wiki/Porting-to-new-MCU-platform#hardware-spi-communication

**Reply by Wosamarak** [reply]
thanks for the explanation
